test/tbb/test_concurrent_queue.cpp

```diff
@@ -281,3 +282,211 @@ TEST_CASE("Test clear and dtor with TrackableItem") {
     test_tracking_dtors_on_clear<oneapi::tbb::concurrent_queue<TrackableItem>>();
     test_tracking_dtors_on_clear<oneapi::tbb::concurrent_bounded_queue<TrackableItem>>();
 }
+
+template<typename CQ, typename T>
+void test_queue_helper() {
+    int size = 5;
+    T vec_1(size, 0), vec_2(size, 0), vec_3(size, 0), vec_4(size, 0);
+    srand(static_cast<unsigned>(time(0)));
+    generate(vec_1.begin(), vec_1.end(), rand);
+    generate(vec_2.begin(), vec_2.end(), rand);
+    generate(vec_3.begin(), vec_3.end(), rand);
+    generate(vec_4.begin(), vec_4.end(), rand);
+
+    CQ q1, q2, q3;
+    CQ q4({vec_1, vec_2, vec_3});
+    q3 = {vec_4, vec_2, vec_3};
+
+    q1 = q3;
+    q2 = std::move(q3);
+    CHECK(q3.empty());
+
+    CHECK(q1 != q4);
+    q1.swap(q4);
+    CHECK(q2 == q4);
+
+    swap(q2, q3);
+    CHECK(q2.empty());
+    CHECK(q3 == q4);
+}
+
+//! basic test for copy, move and swap
+//! \brief \ref error_guessing
+TEST_CASE("Test iterator queue") {
```

Comment:
Could you please change the name of the test case to better explain what is tested.

Reply:
renamed